### PR TITLE
IDV FrontEnd app integration with liveness

### PR DESCRIPTION
**Pull request by vineetkachhawaha**
*Description of changes:*
1. New UI
2. New React APP
3. FrontEnd integration with Amazon Rekognition Face Liveness (https://aws.amazon.com/rekognition/face-liveness/) 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/frontend/src/Components/Liveness.js [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+                                fontWeight={tokens.fontWeights.light}
+                            >
+                                This feature uses Amazon Web Services. Biometric identifiers and biometric information (“biometric data”) may be collected, stored, and used by Amazon Web Services for the purpose of comparing the image of an individual with a stored image for analysis, verification, fraud, and security purposes.
+                            </Heading>      <Heading
+
+                                level={6}
+                                marginBottom={tokens.space.large}
+                                fontWeight={tokens.fontWeights.light}
+                            >
+                                Biometric information that is generated as part of this process will be retained in line with Amazon Web Services privacy policy. You hereby provide your express, informed, written release and consent for Amazon Web Services to collect, use, and store your biometric data as described herein.
```
Comment (dr-natetorious) [inline src/frontend/src/Components/Liveness.js:191]: > Amazon Web Services’ privacy policy will retain generated biometric information from this process. You provide your express, informed, written release and consent for Amazon Web Services to collect, use, and store your biometric data as described herein.

This is more grammatically correct, but it feels out of place. AWS doesn't retain the biometric information. This is the property of the account owner

File: src/frontend/src/Components/SideNavigation.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    API.get("identityverification", "reset-user", options).then(response => {
+        // localStorage.removeItem("pic");
+        navigate("/");
+     });
+  }
+  const { activeHref, handleFollow } = useSideNavigation()
+  // const location = useLocation()
+  let items = [...MAIN_ITEMS]
+
+  // items = items.concat([
```
Comment (dr-natetorious) [inline src/frontend/src/Components/SideNavigation.js:41]: no longer needed?

File: src/frontend/src/Error.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const ErrorMessage = {
+ 'UserAlreadyExists' : 'User already exist. Delete user first to try out this feature',
```
Comment (dr-natetorious) [inline src/frontend/src/Error.js:2]: >The user already exists. Delete the user first to try out this feature.

File: src/frontend/src/Error.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+ 'FaceNotMatchWithIDCard':'Selfie does not match the ID card face image. Please use your own valid ID documents.',
```
Comment (dr-natetorious) [inline src/frontend/src/Error.js:3]: The image from the camera and ID card don’t match. Please use your valid ID documents.

File: src/frontend/src/Error.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+'GenericError': 'Error occur in submitting the form. Please try again later!',
```
Comment (dr-natetorious) [inline src/frontend/src/Error.js:4]: An error occurred while submitting the form. Please try again later.

File: src/frontend/src/Error.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+'ValueError': 'User not exist. Please register user first.',
```
Comment (dr-natetorious) [inline src/frontend/src/Error.js:5]: The user does not exist. Please register the user first.

File: src/frontend/src/Error.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+'LiveNessResultThreshold':'Liveness confidence score is less compare to user selected confidence score. Please do the liveness check again'
```
Comment (dr-natetorious) [inline src/frontend/src/Error.js:6]: The Liveness confidence score is less than the user-selected confidence threshold. Please reattempt the liveness check again.

File: src/frontend/src/Pages/Home.js [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+function Home() {
+
+    const { tokens } = useTheme();
+    const navigate = useNavigate()
+
+    const items = [
+        {
+            title: 'Grow users faster',
+            description:
+                'Convert more visitors into customers by decreasing onboarding time and increasing user convenience. With Amazon Rekognition, you can verify users in seconds online anywhere in the world, and scale from hundreds to millions of identity verifications per hour. Users can now access your services online without having to visit in-person.',
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Home.js:26]: > Convert more visitors into customers by decreasing onboarding time and increasing user convenience. With Amazon Rekognition, you can verify users in seconds online anywhere in the world and scale from hundreds to millions of identity verifications per hour. Users can now access your services online without having to visit in person.

The last sentence seems out of place

File: src/frontend/src/Pages/Home.js [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            title: 'Reduce fraud',
+            description:
+                'Strengthen your fraud prevention capabilities by complementing passwords-based authentication with online visual identity verification. Guard against fraudulent account openings or transactions by comparing user’s selfie picture with identity document picture or your collection of existing users’ pictures.',
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Home.js:31]: >Strengthen your fraud prevention capabilities by complementing password-based authentication with online visual identity verification. Guard against fraudulent account openings or transactions by comparing a user's photo with an identity document picture or your existing user image collection.

File: src/frontend/src/Pages/Home.js [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            title: 'Lower costs and overheads',
+            description:
+                "Reduce the time and cost of in-person identity verification by using Amazon Rekognition pre-trained and customizable APIs. With Amazon Rekognition, you can onboard and authenticate users online without building and managing your own ML infrastructure.",
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Home.js:36]: > Reduce the time and cost of in-person identity verification by using Amazon Rekognition pre-trained and customizable APIs. With Amazon Rekognition, you can onboard and authenticate users online without building and managing your ML infrastructure.

File: src/frontend/src/Pages/Home.js [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+                        <Link color="white"
+                            href="https://github.com/aws-samples/rekognition-identity-verification"
+                            isExternal={true}
+                        >
+                            Deploy Solution Template
+                        </Link>
+                    </Button>
+                    <Button variation="primary" onClick={() => navigate(`/login`)} color="black">
+
+                        Try Out
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Home.js:68]: Try it Out

File: src/frontend/src/Pages/Home.js [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+                <Card>
+                    <Flex alignItems="flex-start"
+                        backgroundColor={tokens.colors.background.secondary}
+                        marginTop={tokens.space.large}
+                        direction={{ base: 'column', large: 'row' }}>
+                        <Text
+                            as="span"
+                            color="black"
+                        >
+                            In-person user identity verification is slow to scale, costly, and high friction for users. Machine learning powered facial biometrics can enable online user identity verification. Amazon Rekognition offers pre-trained facial recognition and analysis capabilities that you can quickly add to your user onboarding and authentication workflows to verify opted-in users' identity online. No machine learning expertise is required. With Amazon Rekognition, you can onboard and authenticate users in seconds while detecting fraudulent or duplicate accounts. As a result, you can grow users faster, reduce fraud, and lower user verification costs.
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Home.js:84]: > In-person user identity verification is slow to scale, costly, and has high friction. Machine learning-powered facial biometrics can enable online user identity verification. Amazon Rekognition offers pre-trained facial recognition and analysis capabilities to quickly add to user onboarding and authentication workflows to verify opted-in users' identities online. No machine learning expertise is required. With Amazon Rekognition, you can onboard and authenticate users in seconds while detecting fraudulent or duplicate accounts. As a result, you can grow users faster, reduce fraud, and lower user verification costs.

File: src/frontend/src/Pages/RegisterWithIdCard.js [hunk 13]
```diff
@@ -1,206 +1,594 @@
...
+
+        }
+        const objectUrl = URL.createObjectURL(event.target.files[0])
+        setPreview(objectUrl)
+        // free memory when ever this component is unmounted
+        // return () => URL.revokeObjectURL(objectUrl)
+
+    }
 
     // useEffect(() => {
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/RegisterWithIdCard.js:137]: Intentional?

File: src/frontend/src/Pages/SignIn.js [hunk 14]
```diff
@@ -0,0 +1,270 @@
...
+    }
+
+
+    const handleNextSubmit = () => {
+        setError({ 'idError': false })
+        if (!errorCheck()) {
+            setHasFormError('')
+            const requestData = {
+                body: { "UserId": id }, // replace this with attributes you need
+                headers: { "Content-Type": "application/json" }, // OPTIONAL
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/SignIn.js:80]: why is it optional?

File: src/frontend/src/App.js [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+
+
+Auth.configure({
+    "Auth": {
+        "identityPoolId": process.env.REACT_APP_IDENTITYPOOL_ID,
+        "region": process.env.REACT_APP_REGION,
+        "userPoolId": process.env.REACT_APP_USERPOOL_ID,
+        "mandatorySignIn": false,
+        "userPoolWebClientId": process.env.REACT_APP_WEBCLIENT_ID
+        // "identityPoolId": "us-east-1:763cb669-b748-498a-ae6b-4cabfb0e457e",
```
[the remark this answers is not in the document]
Comment (vineetkachhawaha) [reply src/frontend/src/App.js:24]: Env variable need to be set for AUTH configure and API configure.

File: src/frontend/src/Components/Header.js [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
...MS4zIDIuMSAxLjMgNC4xdjEuOWMtMS4xLS4yLTIuMS0uNC0zLjEtLjV6bTMuMSA0LjZjMCAuOS0uMSAxLjYtLjMgMi4zLS4yLjYtLjUgMS4yLS45IDEuNy0uNy44LTEuNiAxLjQtMi41IDEuNy0xIC4zLTEuOS41LTIuNy41LTEuMiAwLTIuMS0uMy0yLjgtLjktLjctLjYtMS0xLjUtMS0yLjggMC0xLjMuNC0yLjMgMS4zLTMgLjktLjcgMi4yLTEuMSA0LTEuMS44IDAgMS42LjEgMi41LjIuOS4xIDEuNy4zIDIuNC40djF6Ii8+PC9kZWZzPjxkZWZzPjxwYXRoIGlkPSJTVkdJRF8yXyIgZD0iTTcxIDM1LjNjLTkuOCA0LjEtMjAuNSA2LjEtMzAuMiA2LjEtMTQuNCAwLTI4LjMtMy45LTM5LjYtMTAuMy0xLS42LTEuNy40LS45IDEuMkMxMC44IDQxLjUgMjQuNiA0NyAzOS45IDQ3YzEwLjkgMCAyMy42LTMuNCAzMi40LTkuNyAxLjUtMSAuMi0yLjYtMS4zLTJ6bTguNy00LjljLTEtMS4yLTkuMi0yLjItMTQuMiAxLjMtLjguNS0uNiAxLjMuMiAxLjIgMi44LS4zIDkuMS0xLjEgMTAuMi4zIDEuMSAxLjQtMS4yIDcuMi0yLjMgOS44LS4zLjguNCAxLjEgMS4xLjUgNC43LTMuOCA2LTExLjkgNS0xMy4xeiIvPjwvZGVmcz48dXNlIHhsaW5rOmhyZWY9IiNTVkdJRF8xXyIgb3ZlcmZsb3c9InZpc2libGUiIGZpbGw9IiNGRkYiLz48dXNlIHhsaW5rOmhyZWY9IiNTVkdJRF8yXyIgb3ZlcmZsb3c9InZpc2libGUiIGZpbGw9IiNGODk5MUQiLz48Y2xpcFBhdGggaWQ9IlNWR0lEXzNfIj48dXNlIHhsaW5rOmhyZWY9IiNTVkdJRF8xXyIgb3ZlcmZsb3c9InZpc2libGUiLz48L2NsaXBQYXRoPjxjbGlwUGF0aCBpZD0iU1ZHSURfNF8iIGNsYXNzPSJzdDAiPjx1c2UgeGxpbms6aHJlZj0iI1NWR0lEXzJfIiBvdmVyZmxvdz0idmlzaWJsZSIvPjwvY2xpcFBhdGg+PC9zdmc+'} alt="recipe thumbnail" />
```
[the remark this answers is not in the document]
Comment (vineetkachhawaha) [reply src/frontend/src/Components/Header.js:24]: It can be. But it is for the logo.

File: src/frontend/src/Components/Liveness.js (hunk 1 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Components/Liveness.js:191]: updated

File: src/frontend/src/Components/SideNavigation.js (hunk 2 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Components/SideNavigation.js:41]: removed

File: src/frontend/src/Error.js (hunk 3 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Error.js:2]: Done

File: src/frontend/src/Error.js (hunk 4 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Error.js:3]: done

File: src/frontend/src/Error.js (hunk 5 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Error.js:4]: done

File: src/frontend/src/Error.js (hunk 7 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Error.js:6]: Done

File: src/frontend/src/Pages/Home.js (hunk 8 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Pages/Home.js:26]: I removed this from the page

File: src/frontend/src/Pages/Home.js (hunk 11 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Pages/Home.js:68]: done

File: src/frontend/src/Pages/Home.js (hunk 12 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Pages/Home.js:84]: Updated with the latest wording from https://aws.amazon.com/rekognition/face-liveness/

Comment (vineetkachhawaha) [reply src/frontend/src/Pages/Home.js:84]: Done

File: src/frontend/src/Pages/RegisterWithIdCard.js (hunk 13 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Pages/RegisterWithIdCard.js:137]: Removed

File: src/frontend/src/App.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (dr-natetorious) [reply src/frontend/src/App.js:24]: I'm referring to the commented out section

File: src/frontend/src/Components/Liveness.js [hunk 17]
```diff
@@ -0,0 +1,373 @@
...
+    const [imageData, setImageData] = React.useState(null);
+    const [userSelectedConfidence, setuserSelectedConfidence] = React.useState(80);
+    const [open, setOpen] = React.useState(true);
+
+    const [error, setError] = useState(undefined);
+    // const [checkFailed, setCheckFailed] = useState(false);
+
+    const closeModal = () => setOpen(false);
+
+    // const drawRectangle = (image) => {
```
[the remark this answers is not in the document]
Comment (dr-natetorious) [reply src/frontend/src/Components/Liveness.js:38]: remove the commented out section

**Review by dr-natetorious — APPROVED** [review]

**Pull request merged**